javascript/frameworks/cap/lib/advanced_security/javascript/frameworks/cap/CAPCqlInjectionQuery.qll

```diff
@@ -0,0 +1,216 @@
+
+  string toString() { result = super.toString() }
+}
+
+/**
+ * An await expression that has as its operand a CQL clause that includes a
+ * string concatenation operation.
+ */
+class AwaitCqlClauseWithStringConcatParameter extends CqlInjectionSink {
+  DataFlow::Node queryParameter;
+  DataFlow::Node query;
+  CqlClauseWithStringConcatParameter cqlClauseWithStringConcat;
+
+  AwaitCqlClauseWithStringConcatParameter() {
+    exists(AwaitExpr await |
+      this = await.flow() and
+      await.getOperand() = cqlClauseWithStringConcat.(CqlClause).asExpr()
+    )
+  }
+
+  override DataFlow::Node getQuery() { result = cqlClauseWithStringConcat.(CqlClause).flow() }
+}
+
+/**
+ * The first argument passed to the call to `cds.run`, `cds.db.run`, or `srv.run`
+ * whose value is a CQL query object that includes a string concatenation. e.g.
+ * ``` javascript
+ * // 1. CQN object constructed from Fluent API
+ * const query = SELECT.from`Entity1`.where("ID=" + id);
+ * cds.run(query);
+ *
+ * // 2. CQN object parsed from a string
+ * const query = cds.parse.cql("SELECT * from Entity1 where ID =" + id);
+ * cds.run(query);
+ *
+ * // 3. An unparsed CQL string (only valid in old versions of CAP)
+ * const query = "SELECT * from Entity1 where ID =" + id;
+ * Service2.run(query);
+ * ```
+ * The `getQuery/0` member predicate gets the `query` argument of the above calls
+ * to `run`.
+ */
+class StringConcatParameterOfCqlRunMethodQueryArgument extends CqlInjectionSink {
+  CqlRunMethodCall cqlRunMethodCall;
+
+  StringConcatParameterOfCqlRunMethodQueryArgument() {
+    this = cqlRunMethodCall.getAQueryParameter()
+  }
+
+  override DataFlow::Node getQuery() { result = this }
+}
+
+/**
+ * A CQL shortcut method call (`read`, `create`, ...) parameterized with a string
+ * concatenation expression. e.g.
+ * ``` javascript
+ * cds.read("Entity1").where(`ID=${id}`); // Notice the surrounding parentheses!
+ * cds.create("Entity1").entries({id: "" + id});
+ * cds.update("Entity1").set("col1 = col1" + amount).where("col1 = " + id);
+ * cds.insert("Entity1").entries({id: "" + id});
+ * cds.upsert("Entity1").entries({id: "" + id});
+ * cds.delete("Entity1").where("ID =" + id);
+ * ```
+ */
+class CqlShortcutMethodCallWithStringConcat instanceof CqlShortcutMethodCall {
+  DataFlow::Node stringConcatParameter;
+
+  CqlShortcutMethodCallWithStringConcat() {
+    stringConcatParameter = super.getAQueryParameter() and
+    exists(StringConcatenation::getAnOperand(stringConcatParameter))
+  }
+
+  Location getLocation() { result = super.getLocation() }
+
+  string toString() { result = super.toString() }
+
+  DataFlow::Node getStringConcatParameter() { result = stringConcatParameter }
+}
+
+/**
+ * A string concatenation expression included in a CQL shortcut method call. e.g.
+ * ``` javascript
+ * cds.read("Entity1").where(`ID=${id}`); // Notice the surrounding parentheses!
+ * cds.create("Entity1").entries({id: "" + id});
```

Comment:
I don't think the `entries` cases in this qldoc are actually covered by the class.

Reply:
This is related to our hole in the coverage for `create`, thus will be addressed together.

Reply:
Sorry for the confusion, it's already covered: See [this comment])https://github.com/advanced-security/codeql-sap-js/pull/200#discussion_r2180942518).

Reply:
See comment above.

Reply:
Removed test cases where a call to `entries` is involved in [9101ff3](https://github.com/advanced-security/codeql-sap-js/pull/200/commits/9101ff31a4289d8c7d98a69f1bc42424689ef93f), and got rid of the related propagation step in [a763874](https://github.com/advanced-security/codeql-sap-js/pull/200/commits/a763874f95d6d4abae38eb7670da03e104f34b01).